Celeste.Mod.mm/Patches/Session.cs

```diff
@@ -1,17 +1,91 @@
 #pragma warning disable CS0626 // Method, operator, or accessor is marked external and has no attributes on it
 
+using Celeste.Mod;
 using Mono.Cecil;
 using Mono.Cecil.Cil;
 using MonoMod;
 using MonoMod.Cil;
 using System;
+using System.Collections.Generic;
+using System.Xml.Serialization;
 
 namespace Celeste {
     public class patch_Session {
+
+        public class Slider {
+            private patch_Session _Session;
+
+            public readonly string Name;
+            internal float _Value;
+
+            internal Slider(patch_Session session, string name, float value = 0f) {
+                _Session = session;
+                Name = name;
+                _Value = value;
+            }
+
+            public float Value {
+                get => _Value;
+                set => _Session.SetSlider(this, value);
+            }
+        }
+
+        /// <summary>
+        /// Used internally for serialization.
+        /// </summary>
+        public static class SerializationOnly {
+            [Serializable]
+            public struct Slider {
+                [XmlAttribute]
+                public string Name;
+                [XmlAttribute]
+                public float Value;
+            }
+        }
+
+        [XmlIgnore]
+        private Dictionary<string, Slider> _Sliders;
+
+        [XmlIgnore]
+        public IReadOnlyDictionary<string, Slider> Sliders;
+
+        /// <summary>
+        /// Used internally for serialization; getting or setting this will cause issues.
+        /// Instead, use <see cref="Sliders" />.
+        /// </summary>
+        ///
+        /// NOTE: this cannot be private or [Obselete] because either of those would break serialization.
+        [XmlArray("Sliders")]
+        public SerializationOnly.Slider[] SlidersSerializationOnly {
```

Comment:
`[EditorBrowsable(EditorBrowsableState.Never)]`?

Reply:
oh true, forgot about that